app/configs.py

```diff
@@ -1,45 +1,131 @@
+
+        # properties map
+        self._properties_map = {
+            "DEFAULT_LOG_LEVEL": default_log_level,
+            "LOG_LEVEL_TABLE": log_level_table,
+            "BOOT_DIR": boot_dir,
+            "ETC_DIR": etc_dir,
+            "FSTAB_FILE": fstab_file,
+            "ECU_INFO_FILE": ecu_info_file,
+            "PASSWD_FILE": passwd_file,
+            "GROUP_FILE": group_file,
+            "BOOT_OTA_PARTITION_FILE": ota_partition_folder,
+            "OTA_STATUS_FNAME": "status",
+            "OTA_VERSION_FNAME": "version",
+            "LOG_FORMAT": "[%(asctime)s][%(levelname)s]-%(filename)s:%(funcName)s:%(lineno)d,%(message)s",
+            "MOUNT_POINT": mount_point,
+        }
+
+    def __getattr__(self, name: str):
+        if name not in self._properties_map:
+            raise AttributeError(f"config option {name} not found")
+        else:
+            return self._properties_map[name]
+
+
+class GrubControlConfig(_BaseConfig):
+    """
+    x86-64 platform, using grub
+    """
+
+    PLATFORM = "grub"
+
+    def __init__(self):
+        super().__init__()
+
+        self.grub_dir = self.BOOT_DIR / "grub"
+        self.grub_cfg_file = self.grub_dir / "grub.cfg"
+        self.custom_cfg_file = self.grub_dir / "custom.cfg"
+        self.default_grub_file = self.grub_dir / "default/grub"
+
+        self._properties_map.update(
+            {
+                "GRUB_DIR": self.grub_dir,
+                "GRUB_CFG_FILE": self.grub_cfg_file,
+                "CUSTOM_CFG_FILE": self.custom_cfg_file,
+                "DEFAULT_GRUB_FILE": self.default_grub_file,
+            }
+        )
+
+
+class CBootControlConfig(_BaseConfig):
+    """
+    NOTE: only for tegraid:0x19, jetson xavier platform
+    """
+
+    PLATFORM = "cboot"
+
+    def __init__(self):
+        super().__init__()
+
+        self.extlinux_file = self.BOOT_DIR / "extlinux/extlinux.conf"
+
+        self._properties_map.update(
+            {
+                "EXTLINUX_FILE": self.extlinux_file,
+                "SLOT_IN_USE_FILE": self.BOOT_DIR / "ota-status/slot_in_use",
+                "OTA_STATUS_DIR": self.BOOT_DIR / "ota-status",
+                "LINUX": self.BOOT_DIR / "Image",
+                "INITRD": self.BOOT_DIR / "initrd",
+                "FDT": self.BOOT_DIR / "tegra194-rqx-580.dtb",
+                "EXTRA_CMDLINE": "console=ttyTCU0,115200n8 console=tty0 fbcon=map:0 net.ifnames=0",
+            }
+        )
+
+
+# helper function to detect platform
+def _detect_platform():
+    if platform.machine() == "x86_64" or platform.processor == "x86_64":
+        return "grub"
+    elif platform.machine() == "aarch64" or platform.processor == "aarch64":
+        return "cboot"
+
+
+def create_config(platform):
+    if platform == "grub":
+        return GrubControlConfig()
+    elif platform == "cboot":
+        return CBootControlConfig()
+
+
+cboot_cfg = CBootControlConfig()
+grub_cfg = GrubControlConfig()
+
+Config = create_config(_detect_platform())
```

Comment:
cboot_cfg is an instance, grub_cfg is also an instance, and Config is also an instance.
Only one instance is required I think.

Reply:
`Config` should be `config` since this `Config` is not Class name, this is instance (=variable).

Reply:
> cboot_cfg is an instance, grub_cfg is also an instance, and Config is also an instance.
> Only one instance is required I think. 
Oh cboot_cfg and grub_cfg are specifically for platform specific codes, like I use grub_cfg for grub_control.py, and cboot_cfg for extlinux.cfg, to avoid using wrong configs for platform specific codes.

Reply:
OK, `create_config` should be return either cboot_cfg intance or grub_cfg instance, not to create new instance.